charts/snapshot-validation-webhook/templates/networkpolicy.yaml

```diff
@@ -0,0 +1,17 @@
+{{- if .Values.NetworkPolicy.enabled -}}
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: {{ include "snapshot-validation-webhook.fullname" . }}
+  labels:
+    {{- include "snapshot-validation-webhook.labels" . | nindent 4 }}
+spec:
+  podSelector:
+    matchLabels:
+      {{- include "snapshot-validation-webhook.selectorLabels" . | nindent 6 }}
+  ingress:
+    - from:
+      - podSelector:
+          matchLabels:
+            {{- include "snapshot-validation-webhook.selectorLabels" . | nindent 12 }}
```

Comment:
I don't quite get what's the point of this network policy? It allows for the pod to talk with other instances of the same deployment, but I don't see how thats useful. The only thing that should have access to the pod is the kube-apiserver (or whatever component actually calls the webhook), and maybe some monitoring.

There is also literally nothing to protect here, apart from maybe some DOS concerns by someone overloading the webhook...

Reply:
The purpose is for clusters that utilize network policies that block all ingress by default (zero trust) the helm tests will fail without a network policy in place that allows it. The helm test pod is making a connection to the webhook pod via curl to perform the test. Without the network policy in place the tests fail for us. In our case, we utilize automated helm installs/updates and run helm tests provided by charts as the default behavior.

Reply:
Well, I guess it can't hurt.